### PR TITLE
tests: Skip 'responses' test if not available on older python

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Review by perlpunk — APPROVED** [review]

**Review by kalikiana — CHANGES_REQUESTED** [review]

File: tests/test_utils.py [hunk 1]
```diff
@@ -79,12 +89,14 @@ def test_walk(data, result):
     assert result == ret
 
 
-@responses.activate(registry=registries.OrderedRegistry)
-def test_retry3():
-    rsp1 = responses.add(responses.GET, "http://host.some", status=503)
-    rsp2 = responses.add(responses.GET, "http://host.some", status=503)
-    rsp3 = responses.add(responses.GET, "http://host.some", status=200)
+if has_registries:
+
```
[the remark this answers is not in the document]
Comment (okurz) [reply tests/test_utils.py:93]: sounds nice but as we need the object "registries.OrderedRegistry" in the next line's fixture I don't see how we can use this approach.

[the remark this answers is not in the document]
Comment (mimi1vx) [reply tests/test_utils.py:93]: @kalikiana @okurz https://docs.pytest.org/en/7.1.x/how-to/skipping.html

**Review by Martchus — APPROVED** [review]

File: tests/test_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (okurz) [reply tests/test_utils.py:93]: mimi1vx already approved but the one other option I see is to move the corresponding test function into another file and then execute `registries = pytest.importorskip("responses.registries")`

[the remark this answers is not in the document]
Comment (kalikiana) [reply tests/test_utils.py:93]: I would have thought you can still skip it, maybe move the fixture into the function.

If you can't I suggest at least checking the version in the import and clarifying what requirement prevents us from running that test.

[the remark this answers is not in the document]
Comment (okurz) [reply tests/test_utils.py:96]: > I would have thought you can still skip it, maybe move the fixture into the function.

I wouldn't know how

> If you can't I suggest at least checking the version in the import and clarifying what requirement prevents us from running that test.

I wouldn't know how to check the version. Many packages have an attribute `__version__` but responses doesn't. But I added a comment to explain that. Is this enough?

[the remark this answers is not in the document]
Comment (kalikiana) [reply tests/test_utils.py:96]: Thanks. That seems clear enough.

**Review by kalikiana — APPROVED** [review]

**Pull request merged**